### DISCUSSION TITLE
Improved Rachford-Rice approach

### DISCUSSION THREAD

**Discussion by pkirkham** [ipqa-research/yaeos#114] [Ideas]
Saw you write about yet another EOS on LinkedIn and had a quick browse through the code. I see that you are using Rachford Rice as the objective function in the code and thought I'd let you know about an alternative:

doi:10.1016/j.fluid.2011.12.005

I came across this last year and implemented it in my own cubic EOS code. One of the issues I was running into with mixtures was there there are so many edge/niche cases where the algorithms just fail, so using a more robust approach seemed like a good idea. You might consider implementing it your own code too, as Rachford Rice is pretty much at the heart of the cubic.

**Comment by fedebenelli** [comment]
Thanks for the idea! Yes, Flash algorithms sometimes have a lot of problems in finding convergence. I will look into it and we might implement it as an option. It seems like a interesting approach, scaling everything with the lightest compound
